File: app/src/main/java/com/kelsos/mbrc/domain/AlbumInfo.java
package com.kelsos.mbrc.domain;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;

@AutoValue
public abstract class AlbumInfo implements Parcelable {
  public abstract String album();

  public abstract String artist();

  public static Builder builder() {
    return new AutoValue_AlbumInfo.Builder();
  }

  @AutoValue.Builder
  public abstract static class Builder {
    public abstract Builder album(String album);

    public abstract Builder artist(String artist);

    public abstract AlbumInfo build();
  }
}

File: app/src/main/java/com/kelsos/mbrc/features/library/AlbumInfo.kt
package com.kelsos.mbrc.features.library

import android.os.Parcel
import android.os.Parcelable

data class AlbumInfo(
  val album: String,
  val artist: String,
  val cover: String?,
) : Parcelable {
  constructor(source: Parcel) : this(
    source.readString() ?: "",
    source.readString() ?: "",
    source.readString(),
  )

  override fun describeContents() = 0

  override fun writeToParcel(
    dest: Parcel,
    flags: Int,
  ) {
    dest.writeString(album)
    dest.writeString(artist)
    dest.writeString(cover)
  }

  companion object {
    @JvmField
    val CREATOR: Parcelable.Creator<AlbumInfo> =
      object : Parcelable.Creator<AlbumInfo> {
        override fun createFromParcel(source: Parcel): AlbumInfo = AlbumInfo(source)

        override fun newArray(size: Int): Array<AlbumInfo?> = arrayOfNulls(size)
      }
  }
}
